### PR TITLE
android: fix compilation with new NDK versions

### PR DESCRIPTION
**Pull request by saghul**

### CODE REVIEW

**Comment by saghul** [comment]
CI: https://ci.nodejs.org/view/libuv/job/libuv-test-commit/349/

**Comment by saghul** [comment]
(Hopefully) fixes https://github.com/libuv/libuv/issues/1417

**Comment by jwinarske** [comment]
This change resolves the build breaks using android-ndk-r15b, although now I see unresolved references when linking against libuv.a.  I don't see these unresolved references when building with android-ndk-r14b.

../src/unix/core.c:677: error: undefined reference to '__cmsg_nxthdr'
../src/unix/process.c:391: error: undefined reference to 'signal'
../src/unix/process.c:399: error: undefined reference to 'sigemptyset'
../src/unix/signal.c:131: error: undefined reference to 'sigfillset'
../src/unix/signal.c:216: error: undefined reference to 'sigfillset'
../src/unix/stream.c:1068: error: undefined reference to '__cmsg_nxthdr'
../src/unix/stream.c:1112: error: undefined reference to 'stderr'
../src/unix/thread.c:532: error: undefined reference to 'pthread_condattr_setclock'
../src/unix/thread.c:642: error: undefined reference to 'pthread_barrier_init'
../src/unix/thread.c:647: error: undefined reference to 'pthread_barrier_destroy'
../src/unix/thread.c:653: error: undefined reference to 'pthread_barrier_wait'
../src/unix/udp.c:381: error: undefined reference to 'in6addr_any'
../src/unix/linux-core.c:259: error: undefined reference to 'sigemptyset'
../src/unix/linux-core.c:260: error: undefined reference to 'sigaddset'
../src/unix/linux-core.c:524: error: undefined reference to 'getpagesize'
../src/unix/fs.c:597: error: undefined reference to 'sendfile64'

**Comment by jwinarske** [comment]
Also in the case of android-ndk-r14b, I can build a dynamic library if I define int main() somewhere...  This would be useful as long as changes are being made for Android.

**Comment by saghul** [comment]
> This change resolves the build breaks using android-ndk-r15b, although now I see unresolved references when linking against libuv.a.

How are you linking you app specifically? The change to universal headers looks suspect here. I got similar problems before, but they should now be addressed. Note that the test runner is linked with libuv.a and the build works.

**Comment by saghul** [comment]
> Also in the case of android-ndk-r14b, I can build a dynamic library if I define int main() somewhere... This would be useful as long as changes are being made for Android.

Not sure I follow. Is this related to this issue? If it's not, please feel free to open a new one, but we're going to need some more information.

**Comment by saghul** [comment]
@bnoordhuis can you PTAL again? I made a couple of (minor) changes, but the extra 👀 won't hurt.

File: android-configure [hunk 1]
```diff
@@ -2,17 +2,20 @@
 
 export TOOLCHAIN=$PWD/android-toolchain
 mkdir -p $TOOLCHAIN
+API=${3:-24}
```
Comment (bnoordhuis) [inline android-configure]: Typo?  Should this be `API=${API:-24}`?

**Comment by jwinarske** [comment]
I'm building a shared library via CMake.  No change to my build config when
switching the libuv.a artifacts (r14b vs r15b).  Libuv.a based on r14b
links fine, r15b is not.  I can provide the verbose build output tomorrow.

I haven't yet attempted to build/execute test runner.


Joel

**Comment by saghul** [comment]
Well, you are not using the build system we provide so 🤷‍♂️  I think you might be running into this problem: https://gitlab.kitware.com/cmake/cmake/issues/16584

File: android-configure (hunk 1 above)
Comment (saghul) [inline android-configure]: no typo, you can now pass a 3rd parameter, which is the API level.

**Comment by saghul** [comment]
> you didn't make changes to the C code, did you? Only moved them around?

Correct.

File: android-configure (hunk 1 above)
Comment (bnoordhuis) [inline android-configure]: Ah, hadn't noticed the $1 and $2 further down.  Objection withdrawn.

**Comment by ADENHAM** [comment]
Praying for you.

**Pull request merged**